### DISCUSSION TITLE
How can I remove a Git branch both locally and remotely?

### DISCUSSION THREAD

**Discussion by SuperSmile0426** [community/community#135553] [Discussions] [answered]
### Select Topic Area

Question

### Body

**Failed Attempts to Remove a Remote Branch:** 

``` bash
 $ git branch -d remotes/origin/bugfix error: branch 'remotes/origin/bugfix' not found. 
 $ git branch -d origin/bugfix error: branch 'origin/bugfix' not found. 
 $ git branch -rd origin/bugfix Deleted remote branch origin/bugfix (was 2a14ef7).  $ git push Everything up-to-date. 
 $ git pull From github.com:gituser/gitproject  * [new branch] bugfix -> origin/bugfix Already up-to-date. 
```
   What is the correct method to delete the `remotes/origin/bugfix` branch both locally and on the remote repository?

**Comment by everforyou** [comment, marked as answer]
**Delete Local Branch**

To delete the local branch, use one of the following commands:

```bash
git branch -d <branch_name>
git branch -D <branch_name>
```

- The `-d` option is an alias for `--delete`, which only deletes the branch if it has already been fully merged in its upstream branch.
- The `-D` option is an alias for `--delete --force`, which deletes the branch "irrespective of its merged status." 
- As of Git v2.3, `git branch -d` learned to honor the `-f` (force) flag.
- You will receive an error if you try to delete the currently selected branch.

**Delete Remote Branch**

As of Git v1.7.0, you can delete a remote branch using:

```bash
git push <remote_name> --delete <branch_name>
```

This might be easier to remember than:

```bash
git push <remote_name> :<branch_name>
```

This syntax was added in Git v1.5.0 "to delete a remote branch or a tag." Starting with Git v2.8.0, you can also use `git push` with the `-d` option as an alias for `--delete`.

**Deleting Remote Branches**

Suppose you’re done with a remote branch — say, you and your collaborators are finished with a feature and have merged it into your remote’s main branch. You can delete a remote branch using the syntax:

```bash
git push <remote_name> :<branch>
```

For example, if you want to delete the `serverfix` branch from the server, you run:

```bash
git push origin :serverfix
```

Finally, execute the following on other machines to propagate changes:

```bash
git fetch --all --prune
```
